File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        // prices = [10,1,5,6,7,1]
        // i,j = 0
        // best = 0
        // 1
        int N = prices.size();
        int i = 0; int j = 1;
        int best =0;
        while (i < N && j < N)  {
            
            while (i < N && j < N && prices[i] < prices[j] ) {
                best = max(best,prices[j] -prices[i]);
                j++;
            }
            i = j;
            j = i+1;
        }
        return best;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    from collections import defaultdict
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        wDict = defaultdict(int)
        L = 0
        res = 0

        for R in range(len(s)):
            Rval = s[R]

            while wDict[Rval] >0:
                wDict[s[L]] -=1
                L+=1
            
            wDict[Rval] =1
            res = max(res,R-L+1)

        return res
